bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/ThingManagerImpl.java

```diff
@@ -187,10 +188,11 @@ public void statusUpdated(Thing thing, ThingStatusInfo statusInfo) {
 
             if (ThingStatus.REMOVING.equals(oldStatusInfo.getStatus())
                     && !ThingStatus.REMOVED.equals(statusInfo.getStatus())) {
-                // only allow REMOVING -> REMOVED transition, all others are illegal
-                throw new IllegalArgumentException(MessageFormat.format(
-                        "Illegal status transition from REMOVING to {0}, only REMOVED would have been allowed.",
-                        statusInfo.getStatus()));
+                // only allow REMOVING -> REMOVED transition, all others are ignored because they are illegal
+                logger.warn(
```

Comment:
If we say that bindings do not stand a chance to prevent thing status updates when REMOVING status is entered, we should probably not even log a warning as this would look like there's a bug in the binding. I'd therefore suggest to change it to debug.